loadbalancer/src/main/java/com/io/loadbalancer/consistent_hashing/Hashing.java

```diff
@@ -0,0 +1,46 @@
+package com.io.loadbalancer.consistent_hashing;
+
+import com.io.loadbalancer.exceptions.NoIMInstanceAvailableException;
+
+import java.util.Arrays;
+import java.util.Random;
+
+public class Hashing {
+
+    int IMMax;
+    PopularityMonitor popularityMonitor;
+    ImageManagerMonitor imageManagerMonitor;
+    Random random = new Random();
+
+    public Hashing(int IMMax, PopularityMonitor popularityMonitor, ImageManagerMonitor imageManagerMonitor) {
+        this.IMMax = IMMax;
+        this.popularityMonitor = popularityMonitor;
+        this.imageManagerMonitor = imageManagerMonitor;
+    }
+
+    private int getCongruence(int number, int modulus) {
+        return (number % modulus + modulus) % modulus;
+    }
+
+    public int getIMMapping(String fileName) throws NoIMInstanceAvailableException {
+        // This function takes only filename as argument - no operations
```

Comment:
Please put this comment above the function

Reply:
Done